pkg/apis/serving/v1/revision_defaults.go

```diff
@@ -30,6 +30,16 @@ func (r *Revision) SetDefaults(ctx context.Context) {
 	r.Spec.SetDefaults(apis.WithinSpec(ctx))
 }
 
+// SetDefaults sets the default value for FailureThreshold and TimeoutSeconds
+func SetDefaults(probe *corev1.Probe) {
+	if probe.FailureThreshold == 0 {
+		probe.FailureThreshold = 3
+	}
+	if probe.TimeoutSeconds == 0 {
+		probe.TimeoutSeconds = 10
+	}
+}
+
```

Comment:
1. I would have added changes to existing SetDefaults function same as [SuccessThreshold](https://github.com/knative/serving/blob/master/pkg/apis/serving/v1/revision_defaults.go#L93)

Reply:
@savitaashture Didn't notice that function. Will try to make changes as per your suggestion and amend to the PR. Thanks